### ISSUE TITLE
v9

### ISSUE THREAD

**Issue by scttcper** [scttcper/ngx-toastr#396]
Still planning out a next version. More drastic changes than before. Likely to happen after the release of angular v6

### support
* angular v6
* rxjs v6

### changes
* no default toast (toast component must be supplied to `module.forRoot().` removes dependency on animations
* Add project dependency `@angular/cdk` and use their portal and overlay
	* what is impact on bundle size?
	* would it be possible to prepend toasts (newestOnTop) currently it only has [append](https://github.com/angular/material2/blob/b4c8443387ab9bb486a2ad19c0c79560db154b73/src/cdk/portal/dom-portal-outlet.ts#L63)   
	* whats left after this is gone?
* confirmation support
* fix timing issue #342

**Comment by yarrgh** [comment]
Should also consider using `@angular/cli` 6.0.0 as it now supports libraries. Also relying on `@angular/cdk` shouldn't impact the library bundle size. It should be a peerDependency and not included with the library and should decrease the overall size. Unless you mean bundle size of individual apps.

I have an early working attempt with new cli and using the cdk: https://github.com/yarrgh/ngx-toastr/tree/cdk-cli6

I have a stackblitz but something is weird with toasts. They haven't been showing up correctly until I toggle the bootstrap 4 setting. https://stackblitz.com/github/qgnivbry

And yes I know, the imports are incorrectly referencing the library (should be `@ngx-toastr/core`). Since the dist directory isn't being committed to the repository, StackBlitz doesn't know where to find the package otherwise.

**Comment by scttcper** [comment]
Cool, I'll check out the demo.

I've put us onto angular 6 with rxjs 6 now. I tried out the library support with an earlier build of angular-cli library support and it didn't seem to do much other than call ng-packagr for you. Still didn't support typescript paths or sub packages or whatever.

Have you read anything about that root injectable?
https://blog.ninja-squad.com/2018/05/04/what-is-new-angular-6/

```ts
@Injectable({
  providedIn: 'root'
})
export class UserService {

}
```
https://angular.io/api/core/Injectable#injectable

**Comment by yarrgh** [comment]
Also wanted your opinion on: 
https://github.com/yarrgh/ngx-toastr/blob/f7003fbf3259fea023a79dca85328b3233b4a52f/projects/ngx-toastr/core/src/lib/toast-token.ts#L10-L13

This uses the `providedIn: root` and the `factory` properties to automatically create the default config. I then made the TOAST_CONFIG token optional in the `ToastrService`:

https://github.com/yarrgh/ngx-toastr/blob/f7003fbf3259fea023a79dca85328b3233b4a52f/projects/ngx-toastr/core/src/lib/toastr.service.ts#L48-L51

My idea is that we could potentially make `forRoot()` optional if you want to just use the defaults.

**Comment by scttcper** [comment]
optional config works for me.

Not sure how to get cdk to do newestOnTop option. I was thinking could change the css to use flexbox and reverse the order, but that seems hacky. I guess we could access the elements and change the order...

**Comment by yarrgh** [comment]
It made sense at the time. It was bundled with the Material library and it didn't make sense to bring in the Material library.

Flexbox might be the simplest/cleanest way of doing it. I'm not sure how easy it would be to access the elements and reverse them with the cdk. Worth investigating if you have any ideas.

One thing I did notice with using the cdk, at least with the way I've implemented it, in the demo, if you switch from a normal toast to a notyf toast while other toasts are visible, it pulls all of the toasts down to the bottom right with the notyf toast. Not sure if that's a huge deal or not. Maybe there's a better way than the way I am doing it.
